python/pyspark/sql/readwriter.py

```diff
@@ -69,20 +82,26 @@ class DataFrameReader(OptionUtils):
     def __init__(self, spark: "SparkSession"):
         self._jreader = spark._jsparkSession.read()
         self._spark = spark
+        self._format: Optional[Union[str, Type[DataSource]]] = None
+        self._schema: Optional[Union[str, StructType]] = None
+        self._options: Dict[str, "OptionalPrimitiveType"] = dict()
```

Comment:
@HyukjinKwon do you think we should define a separate class (like PyDataSourceReader) that contains all the logic related to Python data source? So that we don't need to expose these variables in the DataFrameReader class.

Reply:
All these have to be removed. In theory, `DataFrameReader` should not have any change, and the Python sources are only referenced by `format` short name.

So yes I would like to separate them somewhere to make it easier to remove.

Reply:
Ok sounds good. I will refactor this in the next PR to make the review easier. Added a TODO.